python-avd/pyavd/_eos_designs/structured_config/connected_endpoints/utils.py

```diff
@@ -53,37 +53,52 @@ def _filtered_connected_endpoints(
         Adapters are filtered to contain only the ones connected to this switch.
         """
         filtered_connected_endpoints = EosDesigns._DynamicKeys.DynamicConnectedEndpointsItem.ConnectedEndpoints()
+        for connected_endpoints_key in self.inputs._dynamic_keys.custom_connected_endpoints:
+            self._filtered_adapters(filtered_connected_endpoints, connected_endpoints_key)
+
         for connected_endpoints_key in self.inputs._dynamic_keys.connected_endpoints:
-            for connected_endpoint in connected_endpoints_key.value:
-                filtered_adapters = EosDesigns._DynamicKeys.DynamicConnectedEndpointsItem.ConnectedEndpointsItem.Adapters()
-                for adapter_index, adapter in enumerate(connected_endpoint.adapters):
-                    adapter._internal_data.context = f"{connected_endpoints_key.key}[name={connected_endpoint.name}].adapters[{adapter_index}]"
-                    adapter_settings = self.shared_utils.get_merged_adapter_settings(adapter)
-                    if not adapter_settings.switches or self.shared_utils.hostname not in adapter_settings.switches:
-                        continue
-
-                    # Verify that length of all lists are the same
-                    nodes_length = len(adapter_settings.switches)
-                    endpoint_ports = adapter_settings.endpoint_ports
-                    if len(adapter_settings.switch_ports) != nodes_length or (endpoint_ports and len(endpoint_ports) != nodes_length):
-                        msg = (
-                            f"Length of lists 'switches' ({len(adapter.switches)}), 'switch_ports' ({len(adapter.switch_ports)}), "
-                            f"'endpoint_ports' ({len(endpoint_ports) or '-'}) (if used) did not match on adapter {adapter_index} on"
-                            f" connected_endpoint '{connected_endpoint.name}' under '{connected_endpoints_key.key}'."
-                            " Notice that some or all of these variables could be inherited from 'port_profiles'"
-                        )
-                        raise AristaAvdError(msg)
-
-                    filtered_adapters.append(adapter_settings)
-
-                if filtered_adapters:
-                    # The object was deepcopied inside "get_merged_adapter_settings" so we can modify it here.
-                    connected_endpoint.adapters = filtered_adapters
-                    connected_endpoint._internal_data.type = self.inputs.connected_endpoints_keys[connected_endpoints_key.key].type
-                    filtered_connected_endpoints.append(connected_endpoint)
+            if connected_endpoints_key.key not in self.inputs.custom_connected_endpoints_keys:
```

Comment:
Let's create a shared util instead where we emit the complete list and have filtered away the builtin's shadowed by the custom ones. This util can then be shared by all the places you need to read them.

Reply:
Moved the code.